streams/src/test/java/org/apache/kafka/streams/processor/TopologyBuilderTest.java

```diff
@@ -227,25 +227,27 @@ public void testAddStateStoreWithDuplicates() {
     @Test
     public void testAddStateStore() {
         final TopologyBuilder builder = new TopologyBuilder();
-        List<StateStoreSupplier> suppliers;
 
         StateStoreSupplier supplier = new MockStateStoreSupplier("store-1", false);
         builder.addStateStore(supplier);
-        suppliers = builder.build("X", null).stateStoreSuppliers();
-        assertEquals(0, suppliers.size());
+        builder.setApplicationId("X");
+
+        assertEquals(0, builder.build(null).stateStoreSuppliers().size());
 
         builder.addSource("source-1", "topic-1");
         builder.addProcessor("processor-1", new MockProcessorSupplier(), "source-1");
         builder.connectProcessorAndStateStores("processor-1", "store-1");
-        suppliers = builder.build("X", null).stateStoreSuppliers();
+
+        List<StateStoreSupplier> suppliers = builder.build(null).stateStoreSuppliers();
         assertEquals(1, suppliers.size());
         assertEquals(supplier.name(), suppliers.get(0).name());
     }
 
     @Test
     public void testTopicGroups() {
         final TopologyBuilder builder = new TopologyBuilder();
-
+        builder.setApplicationId("X");
+        builder.addInternalTopic("topic-1x");
```

Comment:
This seems like it's a kind of weird restriction. I guess it'd be odd to use the same name for an internal topic and another, though if you know its going to be prefixed it might not be great to not be able to use that same name.